src/simsopt/objectives/utilities.py

```diff
@@ -66,41 +66,63 @@ def dJ(self):
 
 class QuadraticPenalty(Optimizable):
 
-    def __init__(self, obj, threshold=0.):
+    def __init__(self, obj, cons=0., f="min"):
         r"""
-        A penalty function of the form :math:`\max(J - \text{threshold}, 0)^2` for an underlying objective ``J``.
+        A quadratic penalty function of the form :math:`0.5*f(obj.J() - \text{cons}, 0)^2` for an underlying objective ``obj``
+        and wrapping function `f`. This can be used to implement a barrier penalty function for (in)equality
+        constrained optimization problem. The wrapping function defaults to 'min'.
 
         Args:
             obj: the underlying objective. It should provide a ``.J()`` and ``.dJ()`` function.
-            threshold: the threshold past which values should be penalized.
+            cons: constant
+            f: the function that wraps the difference `obj-\text{cons}`
```

Comment:
How about add in this line that the options are `"min"`, `"max"`, and `"identity"`.